File: src/Java/CreationalPatterns/AbstractFactory/Furniture/src/ConcreteFactories/ClassicFurnitureFactory.java
package ConcreteFactories;

import Interfaces.IFurniture;
import Interfaces.IFurnitureFactory;
import Products.ClassicCabinet;
import Products.ClassicChair;
import Products.ClassicDiningTable;

public class ClassicFurnitureFactory implements IFurnitureFactory {
    @Override
    public IFurniture createCabinet() {
        return new ClassicCabinet();
    }

    @Override
    public IFurniture createChair() {
        return new ClassicChair();
    }

    @Override
    public IFurniture createDiningTable() {
        return new ClassicDiningTable();
    }
}

File: src/PHP/CreationalPatterns/AbstractFactory/Furniture/ConcreteFactories/ClassicFurnitureFactory.php
<?php

namespace CreationalPatterns\AbstractFactory\Furniture\ConcreteFactories;

use CreationalPatterns\AbstractFactory\Furniture\Interfaces\IFurniture;
use CreationalPatterns\AbstractFactory\Furniture\Interfaces\IFurnitureFactory;
use CreationalPatterns\AbstractFactory\Furniture\Products\ClassicCabinet;
use CreationalPatterns\AbstractFactory\Furniture\Products\ClassicChair;
use CreationalPatterns\AbstractFactory\Furniture\Products\ClassicDiningTable;

class ClassicFurnitureFactory implements IFurnitureFactory {
    public function createCabinet(): IFurniture {
        return new ClassicCabinet();
    }

    public function createChair(): IFurniture {
        return new ClassicChair();
    }

    public function createDiningTable(): IFurniture {
        return new ClassicDiningTable();
    }
}